tests/ssl-opt.sh

```diff
@@ -1565,12 +1724,14 @@ run_test    "RC4: server half, client enabled" \
             1 \
             -s "SSL - The server has no ciphersuites in common"
 
+requires_ciphersuite_enabled TLS-RSA-WITH-RC4-128-SHA
 run_test    "RC4: server enabled, client disabled" \
             "$P_SRV force_ciphersuite=TLS-RSA-WITH-RC4-128-SHA" \
```

Comment:
With this PR, this test case is no longer running in configurations that where this RC4 ciphersuite is usable, but disabled by default at runtime due to `MBEDTLS_REMOVE_ARC4_CIPHERSUITES`. This is a quirk of how ciphersuites are detected: `ssl-opt.sh` calls `ssl_client2 help`, which calls `mbedtls_ssl_list_ciphersuites`, which omits ciphersuites removed through `MBEDTLS_REMOVE_ARC4_CIPHERSUITES`.

This test case validates that a client won't try to use RC4 when `MBEDTLS_REMOVE_ARC4_CIPHERSUITES` is enabled, so it should run in configurations where the ciphersuite is available upon request but disabled by default, which includes the default configuration. So I'll change the dependencies and explain.

The same quirk exists for 3DES ciphersuites, but it doesn't make a difference to `ssl-opt.sh` since it never tries to use DES.

Reply:
I pushed a commit that fixes the RC4 test cases: 719a6528348928be2d99aa52575b8bf94b680357.

I'm not very happy with this hack. How about expanding `query_compile_time_config` to list all ciphersuites (not just the ones enabled by default)?

Reply:
> How about expanding query_compile_time_config to list all ciphersuites (not just the ones enabled by default)?

Done in https://github.com/gilles-peskine-arm/mbedtls/tree/ssl-opt-auto-psk-2.28-query_ciphersuites . But I'm not very happy with the way I've done it. It causes `query_compile_time_config` to require functions in libmbedtls (and so as it stands it'll fail in shared library builds). Rather, I should copy the `ciphersuite_definitions` array from `ssl_ciphersuites.c`.